src/backend/dx11/src/device.rs

```diff
@@ -1004,31 +1123,71 @@ impl hal::Device<Backend> for Device {
 
     fn create_descriptor_pool<I>(
         &self,
-        _max_sets: usize,
-        _descriptor_pools: I,
+        max_sets: usize,
+        ranges: I,
     ) -> DescriptorPool
     where
         I: IntoIterator,
         I::Item: Borrow<pso::DescriptorRangeDesc>
     {
-        // TODO: descriptor pool
+        let count = ranges.into_iter().map(|r| r.borrow().count).sum::<usize>() * max_sets;
 
-        DescriptorPool
+        DescriptorPool::with_capacity(count)
     }
 
     fn create_descriptor_set_layout<I, J>(
-        &self, bindings: I, _immutable_samplers: J
+        &self, layout_bindings: I, _immutable_samplers: J
     ) -> DescriptorSetLayout
     where
         I: IntoIterator,
         I::Item: Borrow<pso::DescriptorSetLayoutBinding>,
         J: IntoIterator,
         J::Item: Borrow<Sampler>,
     {
-        // TODO: descriptorsetlayout
+        let mut max_binding = 0;
+        let mut bindings = Vec::new();
+
+        // convert from DescriptorSetLayoutBinding to our own PipelineBinding, and find the higher
+        // binding number in the layout
+        for binding in layout_bindings {
+            let binding = binding.borrow();
+
+            max_binding = max_binding.max(binding.binding as u32);
+
+            bindings.push(PipelineBinding {
+                stage: binding.stage_flags,
+                ty: binding.ty,
+                binding_range: binding.binding..(binding.binding + 1),
+                handle_offset: 0
+            });
+        }
+
+        // we sort the internal descriptor's handle (the actual dx interface) by some categories to
+        // make it easier to group api calls together
+        bindings.sort_unstable_by(|a, b| {
```

Comment:
nit: could be easier to just do sort by key being `(a.ty, a.binding_range.star, a.stage)`.